### PR TITLE
Stats: Updating Stats purchase page

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to https://github.com/Automattic/jetpack/issues/31344.

## Proposed Changes

* refactoring the code to make it more modular and CSS to follow BEM
* adding missing copy and updating styling

## Testing Instructions

* Open the live branch for this PR
* Navigate to `/stats/purchase/:siteSlug?flags=stats/paid-stats`

| Screenshots | | |
| --- | --- | --- |
| SCR-20230704-izov | SCR-20230704-jaom | SCR-20230704-jars |
| SCR-20230704-javz | SCR-20230704-jbag | |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by jsnmoon — COMMENTED** [review]
Just wanted to note a bug where the illustration fails to promptly remove the "Ad" rectangle and the crossed-out chat bubbles when sliding the value from $0 to a non-zero value:

image

Large GIF (1252x758)

Also, matticbot is (repeatedly) reapplying the "Needs Review" label because there are reviewers assigned to the PR :)

**Comment by ederrengifo** [comment]
A few comments:

- [x] The font size for the buttons seems to be smaller than it should
Screen Shot 2023-07-03 at 17 13 20

- [x] When focusing on the button, it shows the original theme color
Screen Shot 2023-07-03 at 17 15 02

- [x] Text shouldn't break into a new line here
Screen Shot 2023-07-03 at 17 15 25

- [x] Background for the commercial site notification should be green
Screen Shot 2023-07-03 at 17 16 19

- [x] Personal site slider should show a yellow notification when the amount is set < $10. Above that it should disappear. Here a reference from the designs:
Screen Shot 2023-07-03 at 17 17 25

File: client/my-sites/stats/stats-purchase/stats-purchase-commercial.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+				<div className={ `${ COMPONENT_CLASS_NAME }__pricing-value` }>
+					<div className={ `${ COMPONENT_CLASS_NAME }__pricing-currency` }>$</div>
+					<div className={ `${ COMPONENT_CLASS_NAME }__pricing-amount` }>{ `${ planValue }` }</div>
+				</div>
+				<div className={ `${ COMPONENT_CLASS_NAME }__pricing-cadency` }>
+					/{ translate( 'month' ) }
+				</div>
+			</div>
+
+			<div className={ `${ COMPONENT_CLASS_NAME }__benefits` }>
```
Comment (kangzj) [inline client/my-sites/stats/stats-purchase/stats-purchase-commercial.tsx:39]: We could potentially reuse the features on the checkout page:

https://github.com/Automattic/wp-calypso/blob/4bed13fb704fcc9652148c540c1cd8750f2616af/client/my-sites/checkout/composite-checkout/lib/get-jetpack-product-features.ts#L207

File: client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx [hunk 2]
```diff
@@ -1,16 +1,25 @@
...
+
 const PersonalPurchase = ( { subscriptionValue, setSubscriptionValue }: PersonalPurchaseProps ) => {
-	// TODO: Add translate calls to copy.
+	const translate = useTranslate();
+
 	const sliderLabel = ( ( props, state ) => {
-		return <div { ...props }>${ state?.valueNow || subscriptionValue }/month</div>;
+		return (
+			<div { ...props }>
+				${ state?.valueNow || subscriptionValue }/{ translate( 'month' ) }
```
Comment (kangzj) [inline client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx]: In the design, we are using yearly prices. Probably @ederrengifo can confirm.

**Review by kangzj — APPROVED** [review]
I'm very happy with what it achieves with a single PR. Only a couple of nitty notes, no blockers. Great job @grzegorz-cp 👍 


----EDIT----
I also noticed the background included in the design is not applied yet:

image

File: client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx (hunk 2 above)
Comment (ederrengifo) [reply client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx]: oh sorry for the confusion here! We initially used yearly prices but after checking with marketing we decide to make it monthly. I can double check that though

File: client/my-sites/stats/stats-purchase/stats-purchase-commercial.tsx (hunk 1 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-purchase/stats-purchase-commercial.tsx:39]: Good idea. I added one missing feature description. 
Should we add the opposite 'no-' features to free stats? I don't think we can the reversed "x" icon.

Comment (kangzj) [reply client/my-sites/stats/stats-purchase/stats-purchase-commercial.tsx:39]: I'm happy as is. Let's consider this later. 👍

**Comment by ederrengifo** [comment]
Thank you for the fixes @grzegorz-cp. A couple of more comments:

- [ ] I noticed that on bigger screens, the background for the right side doesn't adapt to its width, and it shows a white area:

Screen Shot 2023-07-05 at 11 37 18

- [ ] Can we remove the logic to dismiss the yellow notification above $10? I notice it does a weird jumping, and to avoid making it more complex, it's ok if we keep it on screen all the time.

https://github.com/Automattic/wp-calypso/assets/10525007/4db6e0e6-7dff-4705-a7b6-47eec9421e23

- [ ] I'm not sure if it can be done in this PR, but besides what was mentioned about the blurred objects for the bg, can we make animation for the higher amount? Some celebratory animation!

Screen Shot 2023-07-05 at 11 52 05

**Comment by grzegorz-cp** [comment]
Thank you for the comments @ederrengifo and @kangzj. Due to the recent discussions, more changes are coming, so I'd like to merge this PR and iterate on the remaining feedback there.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8352509

Thank you @grzegorz-cp for including a screenshot in the description! This is really helpful for our translators.

**Comment by dlind1** [comment]
Hi @grzegorz-cp !
We have received feedback that in string `This plan is for personal sites only. If your site is used for a commercial activity, {{Button}}you will need to choose a commercial paln{{/Button}}.` there is a typo - `paln` instead of `plan`.
Also, in string `The average person pays %(value)s per month` it sounds like the person is `average`, would it be possible to rephrase it so it's clear that we're talking about the `average amount paid`?
Thanks!

**Comment by grzegorz-cp** [comment]
Thanks for the suggestion @dlind1. I'll see what I can do about the ambiguous phrase. The adjustments will be made in #79045.

**Pull request merged**